pkg/exporter/ipfix_proto.go

```diff
@@ -0,0 +1,498 @@
+// TimeFlowStart -> flowStartSeconds, flowStartNanoseconds
+// TimeFlowEnd -> flowEndSeconds, flowEndNanoseconds
+// Packets     -> packetDeltaCount
+// Interface   -> interfaceName
+
+func sendDataRecordv6(log *logrus.Entry, record *flow.Record, exporter *ipfixExporter.ExportingProcess, templateID uint16) error {
+	dataSet := entities.NewSet(false)
+	err := dataSet.PrepareSet(entities.Data, templateID)
+	if err != nil {
+		log.Errorf("Failed in PrepareSet")
+		return err
+	}
+	elements := make([]entities.InfoElementWithValue, 0)
+
+	err = addElementToTemplate(log, "ethernetType", makeByteFromUint16(record.EthProtocol), &elements)
+	if err != nil {
+		return err
+	}
+	err = addElementToTemplate(log, "flowDirection", makeByteFromUint8(record.Direction), &elements)
+	if err != nil {
+		return err
+	}
+	err = addElementToTemplate(log, "sourceMacAddress", net.HardwareAddr(record.DataLink.SrcMac[:]), &elements)
+	if err != nil {
+		return err
+	}
+	err = addElementToTemplate(log, "destinationMacAddress", net.HardwareAddr(record.DataLink.DstMac[:]), &elements)
+	if err != nil {
+		return err
+	}
+	err = addElementToTemplate(log, "sourceIPv6Address", record.Network.SrcAddr.IP(), &elements)
+	if err != nil {
+		return err
+	}
+	err = addElementToTemplate(log, "destinationIPv6Address", record.Network.DstAddr.IP(), &elements)
+	if err != nil {
+		return err
+	}
+	err = addElementToTemplate(log, "nextHeaderIPv6", makeByteFromUint8(record.Transport.Protocol), &elements)
+	if err != nil {
+		return err
+	}
+	err = addElementToTemplate(log, "sourceTransportPort", makeByteFromUint16(record.Transport.SrcPort), &elements)
+	if err != nil {
+		return err
+	}
+	err = addElementToTemplate(log, "destinationTransportPort", makeByteFromUint16(record.Transport.DstPort), &elements)
+	if err != nil {
+		return err
+	}
+	err = addElementToTemplate(log, "octetDeltaCount", makeByteFromUint64(record.Bytes), &elements)
+	if err != nil {
+		return err
+	}
+	err = addElementToTemplate(log, "flowStartSeconds", makeByteFromUint32(uint32(record.TimeFlowStart.Unix())), &elements)
+	if err != nil {
+		return err
+	}
+	err = addElementToTemplate(log, "flowStartMilliseconds", makeByteFromUint64(uint64(record.TimeFlowStart.UnixMilli())), &elements)
+	if err != nil {
+		return err
+	}
+	err = addElementToTemplate(log, "flowEndSeconds", makeByteFromUint32(uint32(record.TimeFlowEnd.Unix())), &elements)
+	if err != nil {
+		return err
+	}
+	err = addElementToTemplate(log, "flowEndMilliseconds", makeByteFromUint64(uint64(record.TimeFlowEnd.UnixMilli())), &elements)
+	if err != nil {
+		return err
+	}
+	err = addElementToTemplate(log, "packetDeltaCount", makeByteFromUint64(uint64(record.Packets)), &elements)
+	if err != nil {
+		return err
+	}
+	err = addElementToTemplate(log, "interfaceName", []byte(record.Interface), &elements)
+	if err != nil {
+		return err
+	}
+	err = dataSet.AddRecord(elements, templateID)
+	if err != nil {
+		log.WithError(err).Error("Failed in Add Record")
```

Comment:
No need to log errors if we return the error (the function that receives it will log it again later)